### PR TITLE
Create utils.load module

### PR DESCRIPTION
**Pull request by tatiana**
Relates to: #98

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -72,38 +85,39 @@ def test_table(request, sql_server):
...
-        hook.run(f"DROP TABLE IF EXISTS {table.table_name}")
-        hook.run(f"DROP INDEX IF EXISTS unique_index")
-    elif isinstance(hook, SnowflakeHook):
-        hook.run(f"DROP TABLE IF EXISTS {table.table_name}")
-    elif not isinstance(hook, BigQueryHook):
-        # There are some tests (e.g. test_agnostic_merge.py) which create stuff which are not being deleted
-        # Example: tables which are not fixtures and constraints. This is an aggressive approach towards tearing down:
-        hook.run(f"DROP SCHEMA IF EXISTS {table.schema} CASCADE;")
+    if database == Database.BIGQUERY:
+        hook.run(f"DROP TABLE {table.qualified_name()}")
```
Comment (tatiana) [inline conftest.py:111]: Before, we were not dropping BQ tables :see_no_evil: I cleared up our BQ tables today in GCP

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -46,7 +46,12 @@ jobs:
       - run: python -c 'import os; print(os.getenv("GOOGLE_APPLICATION_CREDENTIALS_JSON", "").strip())' > ${{ env.GOOGLE_APPLICATION_CREDENTIALS }}
       - run: sqlite3 /tmp/sqlite_default.db "VACUUM;"
       - run: pip3 install nox
-      - run: nox -s test_examples_by_dependency
+      - run: nox -s test_examples_by_dependency -- --cov=./ --cov-report=xml --cov-branch
```
Comment (utkarsharma2) [inline .github/workflows/ci.yaml:49]: how is this change related to utils.load module?

Comment (tatiana) [reply .github/workflows/ci.yaml:49]: No - that was a bonus improvement, I noticed these tests were not contributing to the test coverage - which affected the coverage of `astro.utils.dependencies`. I could make a separate PR for this, would you prefer?

Comment (kaxil) [reply .github/workflows/ci.yaml:49]: https://github.com/astro-projects/astro/pull/243 -- should ideally take care of this too 🤞

File: src/astro/sql/operators/agnostic_save_file.py [hunk 3]
```diff
@@ -96,45 +94,43 @@ def file_exists(self, output_file_path, output_conn_id=None):
 
     def convert_sql_table_to_dataframe(self):
         input_table = self.input
-        conn_type = BaseHook.get_connection(input_table.conn_id).conn_type
+        database = get_database_from_conn_id(input_table.conn_id)
 
         # Select database Hook based on `conn` type
         hook_kwargs = {
```
[the remark this answers is not in the document]
Comment (tatiana) [reply src/astro/sql/operators/agnostic_save_file.py:100]: @utkarsharma2 It is indeed redundant! That said, it wasn't introduced as part of this PR..!
Since this PR already grew much more than expected, would it be okay if we log a refactor ticket to review the `save_file` implementation?

File: src/astro/sql/operators/agnostic_sql_append.py [hunk 4]
```diff
@@ -72,16 +74,16 @@ def append(
...
         engine = self.get_sql_alchemy_engine()
-        if self.conn_type in ["postgres", "postgresql"]:
+        if self.schema and get_database_name(engine) != Database.SQLITE:
             metadata = MetaData(schema=self.schema)
         else:
             metadata = MetaData()
         # TO Do - fix bigquery and postgres reflection table issue.
         main_table_sqla = SqlaTable(
-            get_table_name(main_table), metadata, autoload_with=engine
+            main_table.table_name, metadata, autoload_with=engine
```
[the remark this answers is not in the document]
Comment (tatiana) [reply src/astro/sql/operators/agnostic_sql_append.py:83]: That was my first attempt when changing this code! :see_no_evil: But it raised some exceptions.
My understanding was that `SqlaTable` expects to receive the schema in the metadata field - but may be wrong.

File: src/astro/sql/operators/sql_decorator.py [hunk 5]
```diff
@@ -231,20 +236,14 @@ def _set_schema_if_needed(self, schema=None):
...
-        return engine
-
-    def _run_sql(self, sql, parameters):
-        engine = self.get_sql_alchemy_engine()
-        conn = engine.connect()
-        if isinstance(sql, str):
-            return conn.execute(text(sql), parameters)
-        else:
-            return conn.execute(sql, parameters)
+        return get_sqlalchemy_engine(self.hook)
```
[the remark this answers is not in the document]
Comment (tatiana) [reply src/astro/sql/operators/sql_decorator.py:239]: Only because the scope of this PR grew too much already, with several other refactorings.

File: src/astro/utils/database.py [hunk 6]
```diff
@@ -10,6 +10,31 @@
...
+    """
+    Given a conn_id, identify the database name.
+
+    :param conn_id: Airflow connection ID
+    :type conn_id: str
+    :return: the database this interface relates to (e.g. Database.SQLITE)
+    :rtype: astro.constants.Database enum item
+    """
+    conn_type = BaseHook.get_connection(conn_id).conn_type
+    type_to_db = {
```
[the remark this answers is not in the document]
Comment (tatiana) [reply src/astro/utils/database.py:23]: Unfortunately `Database(conn_type)` doesn't work due to `google_cloud_platform`, which should be mapped to the `Database.BigQuery`. 

I liked the package `aenum`. Do you think it is well maintained and stable enough? Could we log a ticket to refactor to use it?

File: src/astro/sql/operators/sql_dataframe.py [hunk 7]
```diff
@@ -125,35 +126,33 @@ def get_snow_hook(self, table: Table) -> SnowflakeHook:
...
             hook = self.get_snow_hook(table)
             df = hook.get_pandas_df(
                 "SELECT * FROM IDENTIFIER(%(input_table)s)",
                 parameters={"input_table": table.table_name},
             )
-        elif conn_type == "sqlite":
+        elif database == Database.SQLITE:
             hook = SqliteHook(sqlite_conn_id=table.conn_id, database=table.database)
             engine = hook.get_sqlalchemy_engine()
             df = pd.read_sql_table(table.table_name, engine)
```
[the remark this answers is not in the document]
Comment (tatiana) [reply src/astro/sql/operators/sql_dataframe.py:154]: @utkarsharma2 I think that is a great improvement. Since it is a refactor on a part that was not changed in this PR, would it be okay if we have a separate ticket for it?

**Comment by tatiana** [comment]
@utkarsharma2 thank you very much for the review comments! I added TypeHints to the `load` module and I logged a ticket to address the other great refactoring suggestions:
https://github.com/astro-projects/astro/issues/245

File: src/astro/sql/operators/sql_dataframe.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply src/astro/sql/operators/sql_dataframe.py:154]: Sure, we can do this later.

File: src/astro/sql/operators/agnostic_sql_append.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply src/astro/sql/operators/agnostic_sql_append.py:83]: Please do, I'll pick this later.

File: src/astro/sql/operators/sql_decorator.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply src/astro/sql/operators/sql_decorator.py:239]: We can do this later on. can we add it as part of refactoring ticket?

File: src/astro/sql/operators/agnostic_sql_append.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tatiana) [reply src/astro/sql/operators/agnostic_sql_append.py:83]: I just added! :raised_hands:

File: src/astro/utils/database.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (utkarsharma2) [reply src/astro/utils/database.py:23]: @tatiana I think, we will run into situations where we need to map one db with multiple connection types, it would help to keep our code base clean. However I'm not sure about the stability and maintainability or the package. @kaxil, do you recommend any packages for this situation?

**Review by utkarsharma2 — APPROVED** [review]
